### ISSUE TITLE
Manipulation of (built-in) Assembly objects fails with Gesture navigation

### ISSUE THREAD

**Issue by johnsonm** [FreeCAD/FreeCAD#14139]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

I have found no combination of modifiers by which I can manipulate Assembly objects under constraint with Gesture navigation (my default). If I switch to Touchpad, I can move assembled parts under constraint.

I have not surveyed all of the navigation modes to determine the full set that are compatible with the built-in Assembly workbench, only failed with Gesture and succeeded with Touchpad so far.

If it's only intended to work with certain navigation modes, documentation of that limitation and closing this report as won't fix or not a bug is fine by me.

### Full version info

```shell
OS: Debian GNU/Linux 12 (bookworm) (GNOME/gnome)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37363 (Git)
Build type: Unknown
Branch: main
Hash: bfbe28f7a5e8098f1fe7239cda39697650027ff2
Python 3.11.2, Qt 5.15.8, Coin 4.0.0, Vtk 9.1.0, OCC 7.6.3
Locale: English/United States (en_US)
Installed mods: 
  * Assembly4 0.50.4
  * Curves 0.6.13
  * sheetmetal 0.3.0
  * ExplodedAssembly
  * fasteners 0.4.66
  * MeshRemodel 1.8919.0
  * freecad.gears 1.0.0
  * Pyramids-and-Polyhedrons
  * dxf_library
  * lattice2 1.0.0
  * boltsfc 2022.11.5
  * in3dca-freegrid 2.0.0
  * parts_library
  * Defeaturing 1.2.0
  * A2plus 0.4.60n
  * CurvedShapes 1.0.4
  * Manipulator 1.5.0
  * Pyramids-and-Polyhedrons.backup1696298372.5270553 (Disabled)
  * Assembly3 0.12.2
(Built with `FC_USE_TNP_FIX`)
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
@PaddleStroke FYI

**Comment by luzpaz** [comment]
@Rexbas  any idea why this would happen ?

**Comment by Rexbas** [comment]
So if I understand the issue correctly... You have an assembly, e.g. AssemblyExample.FCStd, then you double click Assembly in the tree view to get in the mode where you can manipulate the assembly. And then the problem is clicking somewhere on a part of the assembly and moving the mouse does not do anything with Gesture navigation style?

I quickly tested all navigation styles and here are my results:
Gesture works with ALT key but it should probably also work without it so this is a bug.
OpenInventor works with SHIFT key, this is by design.
OpenSCAD is bugged with LMB. It sometimes triggers both assembly manipulation and camera rotation, but it works when using the SHIFT key.

I also noticed that for Gesture the rotation mode as pictured in the tool tip ALT + LMB only works for right ALT not for left ALT.

All issues described here can probably be fixed in the navigation styles. I will add it to my long TODO list :smiley: .

**Comment by johnsonm** [comment]
Thank you!

I definitely only used the left ALT and it never occurred to me to try right ALT. I'm right-handed, and the point of using Gesture for me is primarily that I almost exclusively use a touchpad for FreeCAD, so I want to drive the touchpad mostly with my dominant right hand, and use my left hand for modifiers. This means I have no recollection of ever using right CTRL, and recall using left ALT only by typing it to start compose sequences.

In the meantime, this hint will allow me to work in assembly without changing nav style back and forth a lot, until such time as it bubbles up to the top of your TODO list. :grin:

**Comment by sslupsky** [comment]
I agree, it is preferable that you do not have to use the ALT key to select and move an object in Assembly WB.

@johnsonm Either the left or right ALT key works for me (I am running RC2 on macOS).

**Comment by Rexbas** [comment]
I think this cannot be fixed before 1.0. It seems there is a conflict. The OpenSCAD and Gesture navigation style both rotate when pressing LMB. But LMB is also used for assembly manipulation. The navigation styles process the event first and so it never reaches assembly and I can currently not think of a good easy way to temporarily overrule the navigation style behavior from the assembly view provider.

I do have an idea but that is a massive refactor that also involves other navigation issues/feature requests, not going to happen anytime soon (by me at least).

**Comment by furgo16** [comment]
Still relevant

**Comment by sciunto** [comment]
I'm shimming in. I have issues with assembly in the gesture mode. I have hard time trying to manipulate assemblies. Pressing left ALT is mandatory but not always sufficient for non obvious reasons. Most of the time, ALT + mousse gesture does not change anything. Switching to other modes such as solidworks or touchpad provides naturally a motion while clicking on an edge.




OS: Manjaro Linux (i3/i3/xcb)
Architecture: x86_64
Version: 1.1.0dev.14555 (Git shallow) AppImage
Build date: 2025/09/23 18:47:15
Build type: Release
Branch: main
Hash: e9ce06155d8d47aad7d6c079bfa8ba54df2b6081
Python 3.11.13, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, IfcOpenShell 0.8.2, OCC 7.8.1
Locale: French/France (fr_FR)
Navigation Style/Orbit Style/Rotation Mode: SolidWorks/Trackball/Drag at cursor
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/
Logical DPI/Physical DPI/Pixel Ratio: 96/93.2987/1
Installed mods: 
  * FusedFilamentDesign 0.25.350
  * freecad.gears 1.3.0
  * Woodworking-master 1.0.1.39285.1.1.1.2025.08.06
  * Gridfinity 0.11.9
  * Curves 0.6.70
  * fasteners 0.5.41
  * Manipulator 1.6.2

**Comment by luzpaz** [comment]
* Related: #22429

**Comment by sciunto** [comment]
Summary of my investigations with the help of other users.

* problem reproduced with i3 and cinnamon.
* It appears that the main issue comes from the fact that alt + left  mouse button is caught by these two WM. On i3, it might be due to my specific configuration, i do not use default WIN (mod4) as mod key.

On i3, a solution is to specify in i3 config:
```floating_modifier Mod4```
such that the floating mode manipulation is done with the WIN key while other manipulations are done with $mod.

on cinnamon, other users report that this solution works: https://forums.linuxmint.com/viewtopic.php?p=1432810&sid=3ad02cdbe283ec2c907d7ebbb67f6345#p1432810


**Possible improvement for freecad** add the ability to customize alt (and potentially other keys) of the gesture mode (and potentially other modes), such that users can adapt freecad to their desktop environment.

PS: note also that sometime, getting the ability to move an assembly is difficult, like it needs several alt + rh click to activate.

**Comment by FEA-eng** [comment]
I can't test it with touchscreen, but it seems still relevant.

```
OS: Windows 11 build 26200
Architecture: x86_64
Version: 26.3.0dev.20260630 (Git shallow)
Build date: 2026/06/30 20:58:37
Build type: Release
Branch: (HEAD detached at ba406c730)
Hash: ba406c730396ec276004d5ef6c70d96a5a97488b
Python 3.11.14, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, Clipper2 , IfcOpenShell 0.8.2, OCC 7.8.1
Locale: Polish/Poland (pl_PL) [ OS: Polish/Poland (pl_PL) ]
Stylesheet/Theme/QtStyle: FreeCAD.qss/FreeCAD Light/
Navigation Style/Orbit Style/Rotation Mode: Gesture/Rounded Arcball/Window center
Logical DPI/Physical DPI/Pixel Ratio: 96/113.268/1.25
```
